src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java

```diff
@@ -1,19 +1,26 @@
 package uk.gov.beis.els.categories.waterheaters.model;
 
+import io.swagger.v3.oas.annotations.media.Schema;
 import javax.validation.constraints.Digits;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
+import uk.gov.beis.els.api.common.ApiValuesFromLoadProfileList;
+import uk.gov.beis.els.categories.common.LoadProfile;
 import uk.gov.beis.els.categories.common.StandardTemplateForm50Char;
 import uk.gov.beis.els.model.meta.FieldPrompt;
 
+@Schema(name = "Package of a water heater and solar device energy label calculator")
 public class WaterSolarPackagesCalculatorForm extends StandardTemplateForm50Char {
 
   @FieldPrompt("Water heating energy efficiency of water heater (%)")
   @Digits(integer = 3, fraction = 0, message = "Enter the water heating efficiency (%), up to 3 digits long")
+  @Schema(type = "integer")
+  @NotNull
   private String waterHeatingEfficiencyPercentage;
 
   @FieldPrompt("Declared load profile")
   @NotBlank(message = "Select a declared load profile")
+  @ApiValuesFromLoadProfileList(values = {LoadProfile.M, LoadProfile.L, LoadProfile.XL, LoadProfile.XXL})
```

Comment:
Can this be consistent with the other customisers and get the values from the service class?

Reply:
No because the array needs to be a compile-time constant. So the array needs to be initialised in there
See [this](https://www.baeldung.com/java-annotation-attribute-value-restrictions) for the explanation

Reply:
@FernandoRamirez-De-Aguilar You can add an array of allowed load profiles in the service class then do it the same way as the others.

Reply:
thanks
Done